File: Kth Largest Element in an Array/sol2.cpp

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        int max_int = nums[0];
        int min_int = nums[0];
        for(int i=0; i< nums.size(); i++)//find the max and min in nums
        {
            if(nums[i] > max_int)
                max_int = nums[i];
            else if(nums[i] < min_int)
                min_int = nums[i];
        }  
        vector <int> cnt (max_int - min_int+1, 0);
        for(int i=0; i< nums.size(); i++)
        {
        cnt[nums[i]- min_int] ++;
        }  
        int index = 0;
        for(int i=0; i< cnt.size(); i++)
        {
        for(int j=0; j< cnt[i]; j++)
        {
            nums[index] = i + min_int; 
            index++;
        }
        }
        return nums[nums.size()- k];  
        }
};

File: Contains Duplicate II/sol2.py

class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        
        dic = dict()
        # with a hash map
        # key value of the list, value: index of the list
        for i,n in enumerate(nums):
            if n in dic and i-dic[n]<=k: 
                return True
            dic[n] = i
        return False
